packages/lib/src/vis/shared/ZoomSelectionMesh.tsx

```diff
@@ -0,0 +1,63 @@
+import { useThree } from '@react-three/fiber';
+import type { Vector2 } from 'three';
+
+import type { ModifierKey } from '../models';
+import { useAxisSystemContext } from './AxisSystemContext';
+import SelectionMesh from './SelectionMesh';
+import SelectionRect from './SelectionRect';
+import { useMoveCameraTo } from './hooks';
+
+interface Props {
+  modifierKey?: ModifierKey;
+}
+
+function ZoomSelectionMesh(props: Props) {
+  const { modifierKey } = props;
+
+  const { dataToWorld } = useAxisSystemContext();
+  const moveCameraTo = useMoveCameraTo();
+
+  const { width, height } = useThree((state) => state.size);
+  const camera = useThree((state) => state.camera);
+
+  const onSelection = (startPoint: Vector2, endPoint: Vector2) => {
+    if (startPoint.x === endPoint.x && startPoint.y === endPoint.y) {
+      return;
+    }
+
+    const worldStartPoint = dataToWorld(startPoint);
+    const worldEndPoint = dataToWorld(endPoint);
```

Comment:
We work with data-space coordinates in `SelectionMesh` but in the end, we call `dataToWorld` here and in the rect/line selection components, so perhaps we should reconsider or pass coordinates in both spaces?